src/combinatorics/rado/finset_section.lean

```diff
@@ -0,0 +1,62 @@
+/-
+Copyright (c) 2023 Michael Stoll. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Michael Stoll
+-/
+
+import combinatorics.rado.auxiliary
+
+/-!
+# Sections of families of finite sets
+
+Let `F : ι → finset α` be a family of finite sets. A *section* of `F` is a function `f : ι → α`
+such that `f i ∈ F i` for all `i : ι`. We define `finset.is_section F f` to express this property
+and `finset.is_section_on F f s` to express that `f` is a section on the finset `s`.
+
+We then provide some basic API lemmas.
+
+(Note: `section` is a reserved word, so would be problematic as a name.)
+-/
+
+namespace finset
+
+variables {ι : Type*} {α : Type*}
+
+/-- A function `f : ι → α` is a *section* of `F : ι → finset α` if it maps `i` into `F i`. -/
+def is_section (F : ι → finset α) (f : ι → α) : Prop := ∀ i, f i ∈ F i
```

Comment:
I wonder if it makes sense to generalize this to
```suggestion
def is_section (P : ι → α → Prop) (f : ι → α) : Prop := ∀ i, P i (f i)
```
then your `is_section F f` becomes `is_section (λ i, (∈ F i)) f`.

The motivation here is that this works for `set` too.

Reply:
What about defining `is_section` for `F : ι → set α`? Then one can use `is_section (coe ∘ F) f`. (I'd probably still want to define `finset.is_section` to have simpler notation.) There is one lemma, `is_section_on.image_subset_bUnion`, that really needs a family of `finset`s, though.
So I'm not yet convinced...

Reply:
I tried it, and I will make the change. (This will eliminate `finset_section.lean` from the PR.)